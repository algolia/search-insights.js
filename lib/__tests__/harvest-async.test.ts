describe('Before script loads', () => {
  beforeAll(() => {
    (window as any).AlgoliaAnalyticsObject = 'aa';
    window.aa =
      window.aa ||
      function () {
        (window.aa.queue = window.aa.queue || []).push(arguments);
      };
  });

  it('Should create a a function', () => {
    expect(typeof (window as any).aa).toBe('function');
  });

  it('Should create a queue before script is loaded', () => {
    (window as any).aa('init', { random: 'random' });
    expect((window as any).aa.queue).toMatchSnapshot();
  });
});
