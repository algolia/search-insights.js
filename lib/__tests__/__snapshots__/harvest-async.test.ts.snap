// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Before script loads Should create a queue before script is loaded 1`] = `
[
  Arguments [
    "init",
    {
      "random": "random",
    },
  ],
]
`;
